etl/steps/data/meadow/climate/2023-12-20/surface_temperature.py

```diff
@@ -0,0 +1,125 @@
+    ERA5_combine = ds.sel(expver=1).combine_first(ds.sel(expver=5))
+
+    # Select the 't2m' variable from the combined dataset and assign it to 'da'.
+    da = ERA5_combine["t2m"]
+
+    # Convert the temperature values from Kelvin to Celsius by subtracting 273.15.
+    da_degc = da - 273.15
+
+    # Read the shapefile to extract country informaiton
+    snap_geo = paths.load_snapshot("world_bank.zip")
+    shapefile_name = "WB_countries_Admin0_10m/WB_countries_Admin0_10m.shp"
+
+    # Check if the shapefile exists in the ZIP archive
+    with zipfile.ZipFile(snap_geo.path, "r"):
+        # Construct the correct path for Geopandas
+        file_path = f"zip://{snap_geo.path}!/{shapefile_name}"
+
+        # Read the shapefile directly from the ZIP archive
+        shapefile = gpd.read_file(file_path)
+        shapefile = shapefile[["geometry", "WB_NAME"]]
+
+    # Initialize an empty dictionary to store the country-wise average temperature.
+    temp_country = {}
+
+    # Initialize a list to keep track of small countries where temperature data extraction fails.
+    small_countries = []
+
+    # Iterate over each row in the shapefile data.
+    for i in tqdm(range(shapefile.shape[0])):
+        # Extract the data for the current row.
+        data = shapefile[shapefile.index == i]
+
+        # Set the coordinate reference system for the temperature data to EPSG 4326.
+        da_degc.rio.write_crs("epsg:4326", inplace=True)
+
+        try:
+            # Clip the temperature data to the current country's shape.
+            clip = da_degc.rio.clip(data.geometry.apply(mapping), data.crs)
+
+            # Calculate weights based on latitude to account for area distortion in latitude-longitude grids.
+            weights = np.cos(np.deg2rad(clip.latitude))
+            weights.name = "weights"
+
+            # Apply the weights to the clipped temperature data.
+            clim_month_weighted = clip.weighted(weights)
+
+            # Calculate the weighted mean temperature for the country.
+            country_weighted_mean = clim_month_weighted.mean(dim=["longitude", "latitude"]).values
+
+            # Store the calculated mean temperature in the dictionary with the country's name as the key.
+            temp_country[shapefile.iloc[i]["WB_NAME"]] = country_weighted_mean
+
+        except Exception:
+            # If an error occurs (usually due to small size of the country), add the country's name to the small_countries list.
+            small_countries.append(shapefile.iloc[i]["WB_NAME"])
+
+    # Log information about countries for which temperature data could not be extracted.
+    log.info(
+        f"It wasn't possible to extract temperature data for {len(small_countries)} small countries as they are too small for the resolution of the Copernicus data."
+    )
+    # Define the start and end dates
+    start_time = "1950-01-01"
+    end_time = "2023-11-01"
```

Comment:
You would need to manually change this on every update. Instead, you could do:
```
start_time = da_degc["time"].min().dt.date.astype(str).item()
end_time = da_degc["time"].max().dt.date.astype(str).item()
```

Reply:
goood point! thanks for noticing this!